Unable to find a JDBC driver corresponding to the database kind 'postgresql', when camel-quarkus-quartz dependency added

Originially created [Issue 25817](https://github.com/quarkusio/quarkus/issues/25817) in quarkus.

I have a simple quarkus project using mutiny and postgres, which works with Quarkus 2.7.5.Final. Starting from Quarkus 2.8.0.Final, build fails, when I add some camel route that needs quartz endpoint:

```
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:2.9.1.Final:build (default) on project code-with-quarkus: Failed to build quarkus application: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.agroal.deployment.AgroalProcessor#build threw an exception: io.quarkus.runtime.configuration.ConfigurationException: Unable to find a JDBC driver corresponding to the database kind 'postgresql' for the default datasource. Either provide a suitable JDBC driver extension, define the driver manually, or disable the JDBC datasource by adding 'quarkus.datasource.jdbc=false' to your configuration if you don't need it.
[ERROR] 	at io.quarkus.agroal.deployment.AgroalProcessor.resolveDriver(AgroalProcessor.java:330)
[ERROR] 	at io.quarkus.agroal.deployment.AgroalProcessor.getAggregatedConfigBuildItems(AgroalProcessor.java:276)
[ERROR] 	at io.quarkus.agroal.deployment.AgroalProcessor.build(AgroalProcessor.java:82)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:925)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
....

```

It looks like quartz extension requires the postgresql JDBC driver

How to Reproduce?
Unpack attached project [code-with-quarkus.tar.gz](https://github.com/apache/camel-quarkus/files/8804030/code-with-quarkus.tar.gz)

and run ./mvnw clean package
Remove <artifactId>camel-quarkus-quartz</artifactId> dependency to make build work

--- a/docs/modules/ROOT/pages/reference/extensions/quartz.adoc
+++ b/docs/modules/ROOT/pages/reference/extensions/quartz.adoc
@@ -56,6 +56,15 @@ This is the preferred option. Follow the https://quarkus.io/guides/quartz[schedu
 
 Follow the http://www.quartz-scheduler.org/documentation/quartz-1.8.6/configuration/ConfigJDBCJobStoreClustering.html#configure-clustering-with-jdbc-jobstore[Configure Clustering with JDBC-JobStore Guide] with different DS configuration:
 
+- Declare to use `io.quarkus:quarkus-agroal` in `pom.xml` explicitly.
+
+```
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-agroal</artifactId>
+</dependency>
+```
+
 - Use `quartz.properties` to configure the `JobStore`.
 - Define a datasource via quarkus (see the https://quarkus.io/guides/datasource[Quarkus datasource documentation)] and use `CamelQuarkusQuartzConnectionProvider` as the `ConnectionProvider` in `quartz.properties`:
 

--- a/extensions/quartz/deployment/pom.xml
+++ b/extensions/quartz/deployment/pom.xml
@@ -48,10 +48,6 @@
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-quartz</artifactId>
         </dependency>
-        <dependency>
-            <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-agroal-deployment</artifactId>
-        </dependency>
     </dependencies>
 
     <build>

--- a/extensions/quartz/runtime/pom.xml
+++ b/extensions/quartz/runtime/pom.xml
@@ -56,6 +56,7 @@
         <dependency>
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-agroal</artifactId>
+            <optional>true</optional>
         </dependency>
         <dependency>
             <groupId>org.graalvm.nativeimage</groupId>

--- a/extensions/quartz/runtime/src/main/doc/usage.adoc
+++ b/extensions/quartz/runtime/src/main/doc/usage.adoc
@@ -13,6 +13,15 @@ This is the preferred option. Follow the https://quarkus.io/guides/quartz[schedu
 
 Follow the http://www.quartz-scheduler.org/documentation/quartz-1.8.6/configuration/ConfigJDBCJobStoreClustering.html#configure-clustering-with-jdbc-jobstore[Configure Clustering with JDBC-JobStore Guide] with different DS configuration:
 
+- Declare to use `io.quarkus:quarkus-agroal` in `pom.xml` explicitly.
+
+```
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-agroal</artifactId>
+</dependency>
+```
+
 - Use `quartz.properties` to configure the `JobStore`.
 - Define a datasource via quarkus (see the https://quarkus.io/guides/datasource[Quarkus datasource documentation)] and use `CamelQuarkusQuartzConnectionProvider` as the `ConnectionProvider` in `quartz.properties`:
 

